### PR TITLE
Default to stable channel if charm is not available on the expected one.

### PR DESCRIPTION
**Pull request by ktsakalozos**
Not 100% sure this is the right way to handle this error: https://ci.kubernetes.juju.solutions/job/build-and-release-all-charms-and-bundles/261/console

### CODE REVIEW

File: bundle [hunk 1]
```diff
@@ -118,6 +118,14 @@ def version_bundle(bundle, channel):
         if not resp.ok:
             err = "Couldn't query %s %s, is it public?" % (charm, channel)
             raise CharmstoreQueryError(err)
+        if resp.text == "{}":
+            print("Service {} not avilable in channel {}. Using default channel (stable)."
```
Comment (tvansteenburgh) [inline bundle]: available

**Comment by Cynerva** [comment]
-1 to this. It would be fine except for the odd behavior of Juju where `juju deploy <bundle> --channel edge` also tries to deploy all the _charms_ from edge and blows up in situations like this. See error below.

```
$ ./bundle -o bundle-elastic k8s/cdk cni/flannel monitor/elastic --channel edge
Processing fragment k8s/cdk...
Processing fragment cni/flannel...
Processing fragment monitor/elastic...
Service elasticsearch not avilable in channel edge. Using default channel (stable).
Done. Your bundle can be found in /home/gkk/work/kubernetes/bundle-canonical-kubernetes/bundle-elastic

$ charm push bundle-elastic cs:~cynerva/canonical-kubernetes-elastic
url: cs:~cynerva/bundle/canonical-kubernetes-elastic-0
channel: unpublished

$ charm release cs:~cynerva/bundle/canonical-kubernetes-elastic-0 --channel edge
url: cs:~cynerva/bundle/canonical-kubernetes-elastic-0
channel: edge

$ juju deploy cs:~cynerva/bundle/canonical-kubernetes-elastic --channel edge
Located bundle "cs:~cynerva/bundle/canonical-kubernetes-elastic-0"
Deploying charm "cs:~containers/easyrsa-17"
added resource easyrsa
ERROR cannot deploy bundle: cannot resolve URL "cs:elasticsearch-20": cannot resolve URL "cs:elasticsearch-20": charm or bundle not found
```

**Comment by ktsakalozos** [comment]
@Cynerva is always a step ahead! :)

**Comment by tvansteenburgh** [comment]
Nice catch @Cynerva. You said 'odd' behavior but it seems logical to me since you can't specify a channel for each charm in the bundle individually.

**Pull request closed without merging**